src/core.jsx

```diff
@@ -65,8 +75,11 @@ export default (
 
       state = {
         props: {},
+        promiseLoaded: false,
       }
 
+      promiseResolved = () => { this.setState({ promiseLoaded: true }) }
```

Comment:
Indeed: as you wish. 

I like the idea to have a unique pattern that works everytime and to use it.
Plus, I prefer a functional way of programming instead to rely on a `this` object.